### PR TITLE
inkscape: 1.0.2 -> 1.1

### PR DESCRIPTION
**Pull request by jljusten**
###### Motivation for this change
Upstream update.

https://inkscape.org/news/2021/05/24/welcome-inkscape-11/
https://media.inkscape.org/media/doc/release_notes/1.1/Inkscape_1.1.html

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 124333` at a327d40b run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>gnome.gnome-documents</li>
    <li>spring</li>
    <li>springLobby</li>
  </ul>
</details>
<details>
  <summary>47 packages skipped due to time constraints:</summary>
  <ul>
    <li>adapta-gtk-theme</li>
    <li>arc-theme</li>
    <li>arx-libertatis</li>
    <li>asciidoc-full</li>
    <li>asciidoc-full-with-plugins</li>
    <li>bibata-cursors</li>
    <li>bibata-cursors-translucent</li>
    <li>bibata-extra-cursors</li>
    <li>btrbk</li>
    <li>capitaine-cursors</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>inkscape</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `fetchpatch`.
Near pkgs/applications/graphics/inkscape/default.nix:8:3:
```
  |
8 | , fetchpatch
  |   ^
```
</li>
  </ul>
</details>

**Review by jtojnar — CHANGES_REQUESTED** [review]
Some dependencies changed:

- gdl was replaced by new internal library
- gtkspell was replaced by gspell

**Comment by jtojnar** [comment]
Exporting as JPEG fails with:

```
Traceback (most recent call last):
  File "raster_output_jpg.py", line 23, in <module>
    JpegOutput().run()
  File "/nix/store/gqxcnl2k0v4bm0f0pbyr8a6nx2zv44bf-inkscape-1.1/share/inkscape/extensions/inkex/base.py", line 130, in run
    self.load_raw()
  File "/nix/store/gqxcnl2k0v4bm0f0pbyr8a6nx2zv44bf-inkscape-1.1/share/inkscape/extensions/inkex/base.py", line 143, in load_raw
    document = self.load(self.file_io)
  File "/nix/store/gqxcnl2k0v4bm0f0pbyr8a6nx2zv44bf-inkscape-1.1/share/inkscape/extensions/inkex/extensions.py", line 73, in load
    from PIL import Image
ModuleNotFoundError: No module named 'PIL'
```

**Comment by jtojnar** [comment]
Also lib2geom package can finally be used (needs to be updated first):


https://media.inkscape.org/media/doc/release_notes/1.1/Inkscape_1.1.html#Important_changes_for_packagers

Feel free to pick it from https://github.com/NixOS/nixpkgs/pull/50286

**Comment by jljusten** [comment]
@jtojnar: Thanks for all the tips!

I removed gdl, added pillow for jpg export, and changed from gtkspell to gspell.

I tried to test spell check, but I didn't get the right dictionary installed yet, so it's not been verified.

I also pulled in your lib2geom patch, and updated it to the version used in inkscape 1.1. I tested that lib2geom built after this update. Unfortunately, when I tried to add it as a build dep for inkscape, I start seeing undefined references for double_conversion. Maybe you have a suggestion for this?

File: pkgs/development/libraries/lib2geom/default.nix [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+, glib
+, gsl
+, cairo
+, double-conversion
+, gtest
+}:
+
+stdenv.mkDerivation rec {
+  pname = "lib2geom-unstable";
+  version = "2021-01-07";
```
Comment (jtojnar) [inline pkgs/development/libraries/lib2geom/default.nix:16]: Asked upstream to create a tag: https://gitlab.com/inkscape/lib2geom/-/merge_requests/48#note_584705175

File: pkgs/development/libraries/lib2geom/default.nix [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    "-DCMAKE_SKIP_BUILD_RPATH=OFF" # for tests
+    "-DBUILD_SHARED_LIBS=ON"
+  ];
+
+  doCheck = true;
+
+  meta = with stdenv.lib; {
+    description = "Easy to use 2D geometry library in C++";
+    homepage = "https://gitlab.com/inkscape/lib2geom";
+    license = [ licenses.lgpl21 licenses.mpl11 ];
```
Comment (jtojnar) [inline pkgs/development/libraries/lib2geom/default.nix:57]: The license is now called `lgpl21Only`.

**Comment by jtojnar** [comment]
I guess it has something to do with the fact that it is built as static library, needs "-D2GEOM_BUILD_SHARED=ON"
.

**Comment by jtojnar** [comment]
Also something like this:

```patch
--- a/pkgs/applications/graphics/inkscape/default.nix
+++ b/pkgs/applications/graphics/inkscape/default.nix
@@ -3,7 +3,6 @@
 , boost
 , cairo
 , cmake
-, double-conversion
 , fetchurl
 , gettext
 , ghostscript
@@ -21,6 +20,7 @@
 , libexif
 , libpng
 , librevenge
+, lib2geom
 , librsvg
 , libsigcxx
 , libsoup
@@ -83,6 +83,10 @@ stdenv.mkDerivation rec {
       --replace "call('ps2pdf'" "call('${ghostscript}/bin/ps2pdf'"
     patchShebangs share/templates
     patchShebangs man/fix-roff-punct
+
+    # double-conversion is a dependency of 2geom
+    substituteInPlace CMakeScripts/DefineDependsandFlags.cmake \
+      --replace 'find_package(DoubleConversion REQUIRED)' ""
   '';
 
   nativeBuildInputs = [
@@ -101,7 +105,6 @@ stdenv.mkDerivation rec {
   buildInputs = [
     boehmgc
     boost
-    double-conversion
     gettext
     glib
     glib-networking
@@ -114,6 +117,7 @@ stdenv.mkDerivation rec {
     libexif
     libpng
     librevenge
+    lib2geom
     librsvg # for loading icons
     libsigcxx
     libsoup

```

File: pkgs/development/libraries/lib2geom/default.nix (hunk 1 above)
Comment (SuperSandro2000) [inline pkgs/development/libraries/lib2geom/default.nix:16]:
```suggestion
  pname = "lib2geom";
  version = "unstable-2021-01-07";
```

File: pkgs/development/libraries/lib2geom/default.nix [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  cmakeBuildType = "RelWithDebugInfo"; # needed to keep assertions for tests working
+
+  cmakeFlags = [
+    "-DCMAKE_SKIP_BUILD_RPATH=OFF" # for tests
+    "-DBUILD_SHARED_LIBS=ON"
+  ];
+
+  doCheck = true;
+
+  meta = with stdenv.lib; {
```
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/development/libraries/lib2geom/default.nix:54]:
```suggestion
  meta = with lib; {
```

File: pkgs/development/libraries/lib2geom/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/lib2geom/default.nix:57]:
```suggestion
    license = with licenses; [ lgpl21Only mpl11 ];
```

**Comment by jljusten** [comment]
@jtojnar: Yep, your suggested changes allow the lib2geom link to work. Thanks!

Hopefully I addressed all the other suggestions you made too.

I did have a slight concern about the lib2geom version, as I mentioned here: https://gitlab.com/inkscape/lib2geom/-/merge_requests/48#note_584835008

Nevertheless, I upgraded lib2geom to the 1.1 tagged version, even though it doesn't 100% match what the inkscape 1.1 used.

File: pkgs/development/libraries/lib2geom/default.nix [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    gsl
+    cairo
+    double-conversion
+  ];
+
+  checkInputs = [
+    gtest
+  ];
+
+  cmakeBuildType = "RelWithDebugInfo"; # needed to keep assertions for tests working
```
Comment (jtojnar) [inline pkgs/development/libraries/lib2geom/default.nix:46]: This does not seem to be necessary either.

File: pkgs/applications/graphics/inkscape/default.nix [hunk 5]
```diff
@@ -91,6 +83,10 @@ stdenv.mkDerivation rec {
       --replace "call('ps2pdf'" "call('${ghostscript}/bin/ps2pdf'"
     patchShebangs share/templates
     patchShebangs man/fix-roff-punct
+
+    # double-conversion is a dependency of 2geom
```
Comment (jtojnar) [inline pkgs/applications/graphics/inkscape/default.nix:87]: I have opened an upstream bug, could you please add a link to the comment? https://gitlab.com/inkscape/inkscape/-/issues/2508

**Review by jtojnar — APPROVED** [review]

**Comment by jtojnar** [comment]
Great job. Thanks.

**Comment by jljusten** [comment]
@jtojnar: Thanks for all the help with my first nixpkgs change. :)

**Pull request merged**